lblod/app-lpdc-digitaal-loket
# chore: update the NUTS LAU spatial code list

LPDC uses the resources defined in the NUTS code list to relate product
instances to their geographical scope ("Geografisch toepasingsgebied"). These
migrations update the code list to the most recent version.

The order the migrations are executed is important:
- `delete-nuts2021-code-list` deletes all NUTS resources from the public graph
- `import-nuts2024-codelist` inserts the new version of the NUTS code list in
  the public graph
- `add-uuid-to-nuts2024-resources` adds UUIDs for the new resources such that
  they can be retrieved via the frontend

The changes to the code list are documented [here](https://github.com/Informatievlaanderen/ipdc-lpdc/blob/main/CHANGELOG#L61). Summary of the relevant changes:
- Limit codes for Flemish provinces and municipalities
- Removed the hierarchy between codes, no more `skos:narrower` or `skos:broader`
- Added merged municipalities for 2019 and 2025

## Steps to validate usage of updated code list
Note: For the correct (any) options to appear in the geographical scope form
field, the updated management service should be used: https://github.com/lblod/lpdc-management-service/pull/46

1. Navigate to the mock-login page and select an organisation, any organisation
   that has one or more product instances will do.
2. Select a product instance in the list that appears, any should do.
3. Open the "Eigenschappen" tab for the selected product.
4. At the bottom of the section "Bevoegdheid" there should be a field titled
   "Geografisch toepassingsgebied". This field contains values that should match
   the resources in the NUTS code list.
5. Click on the "Product opnieuw bewerken" button at the top. Go past the
   warning that appears by clicking the "Product opnieuw bewerken" button in the
   modal.
6. Find and try to edit the "Geografisch toepassingsgebied" field. The options
   you are presented with should be only those defined in the updated NUTS code
   list. For example, there should be no municipalities in Brussels or
   arrondissementen. But new municipalities that come from the 2025 mergers,
   such as "Pajottegem", should be available.

## Notes
- The `dcterms:spatial` triples for products instances were not updated. This
  will happen in a follow up PR dealing with the municipality mergers.
- Several locations are `skos:exactMatch` with non-existent NUTS codes, this was
  already the case before the changes proposed here.

## Related tickets
LPDC-1261
